ci_framework/roles/build_containers/tasks/main.yml

```diff
@@ -49,15 +49,15 @@
 
 - name: Make sure authfile exists
   when:
-    - cifmw_build_containers_authfile_path is defined
+    - cifmw_build_containers_authfile_path | default(None) != None
```

Comment:
Can't we just pass this in a custom param file?
```YAML
cifmw_build_containers_authfile_path: null
```
meaning it will be "undefined", so it shouldn't hit that block anyway?

Reply:
hej, like what exactly you want to do that?
A variable with null value in Ansible is still a defined variable apprently (when overrriding within a Playbook) – and when supplying `cifmw_build_containers_authfile_path: ~` on Zuul job vars it also had no effect, but now I am unable to reproduce (`Run container_build.sh script` task fails)

Reply:
I think the problem is that the default is set to a file https://github.com/openstack-k8s-operators/ci-framework/blob/6efd05125bc97d371853d71688c25c51b214ba19/ci_framework/roles/build_containers/defaults/main.yml#L22 so when we try to bring the content-provider, we do need to pass it to none. A better approach would be check also check only if push container is set to true.

Reply:
This is what I wrote in the commit message...
I need a `push_container` to be `true`, because I want the images to land in registry deployed with `registry_deploy` role.

Reply:
Hmmm IIUC then this is wrong, we need to have a push_to_local_container when we want to push to the content-provider and push_container when we want to push to upstream and/or downstream... wdyt @cedric?

Reply:
@arxcruz IMO that overcomplicates the role with additional parameter, while it already has too many parameters...

Reply:
now that I think about it.... in the job reproducer, for the content-provider support, we use the local registry as deployed by that `registry_deploy` role and I didn't have any issue when it came to push content to it.... Fun.

Reply:
probably you already had the file locally in your computer for some other reason, so it passes, but in the clean vm it is failing :)

Reply:
According to https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/reproducer/templates/play.yml.j2#L2-L78 the reproducer uses `operator_build`, not `build_containers` – I am not sure about the difference thought...